openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -21,450 +21,428 @@
 
-
-def _process_media_data(media_data):
-    foreign_id = media_data.get("pageid")
-    logger.debug(f"Processing page ID: {foreign_id}")
-    media_info = _get_image_info_dict(media_data)
-    valid_mediatype = _check_mediatype(media_info)
-    if not valid_mediatype:
-        return None
-    license_info = _get_license_info(media_info)
-    if license_info.url is None:
-        return None
-    media_url = media_info.get("url")
-    if media_url is None:
-        return None
-    creator, creator_url = _extract_creator_info(media_info)
-    title = _extract_title(media_info)
-    filesize = media_info.get("size", 0)  # in bytes
-    filetype = _extract_file_type(media_info)
-    parsed_data = {
-        "media_url": media_url,
-        "foreign_landing_url": media_info.get("descriptionshorturl"),
-        "foreign_identifier": foreign_id,
-        "license_info": license_info,
-        "creator": creator,
-        "creator_url": creator_url,
-        "title": title,
-        "filetype": filetype,
-        "filesize": filesize,
+class WikimediaCommonsDataIngester(ProviderDataIngester):
+    providers = {
+        "image": prov.WIKIMEDIA_DEFAULT_PROVIDER,
+        "audio": prov.WIKIMEDIA_AUDIO_PROVIDER,
     }
+    endpoint = f"https://{HOST}/w/api.php"
+    headers = {"User-Agent": prov.UA_STRING}
+
+    # The batch_limit applies to the number of pages received by the API, rather
+    # than the number of individual records. This means that for Wikimedia Commons
+    # setting a global `ingestion_limit` will still limit the number of records, but
+    # the exact limit may not be respected.
+    batch_limit = 250
+
+    def __init__(self, conf: dict = None, date: str | None = None):
+        self.start_timestamp, self.end_timestamp = self.derive_timestamp_pair(date)
+        self.continue_token = {}
+
+        super().__init__(conf, date)
+
+    def get_next_query_params(self, prev_query_params, **kwargs):
+        return {
+            "action": "query",
+            "generator": "allimages",
+            "gaisort": "timestamp",
+            "gaidir": "newer",
+            "gailimit": self.batch_limit,
+            "prop": "imageinfo|globalusage",
+            "iiprop": "url|user|dimensions|extmetadata|mediatype|size|metadata",
+            "gulimit": self.batch_limit,
+            "gunamespace": 0,
+            "format": "json",
+            "gaistart": self.start_timestamp,
+            "gaiend": self.end_timestamp,
+            **self.continue_token,
+        }
+
+    def get_media_type(self, record):
+        """Get the media_type of a parsed Record"""
+        return record["media_type"]
+
+    def get_response_json(self, query_params):
+        """
+        Overrides the parent function to make multiple requests until
+        we see "batchcomplete", rather than a single request to the
+        endpoint. This ensures that global usage data used for calculating
+        popularity is tabulated correctly.
```

Comment:
I went back and forth on this in the initial implementation and decided to override `get_response_json` because I thought it was more likely for `get_batch` to have future changes (for example we're looking at addressing error handling there right now). Good point about using `super().get_response_json` here! Done ✔️

Reply:
Sounds great, and I love the changes you made there 💯